### PR TITLE
doc: introduce Split Device Model feature for ACRN

### PR DESCRIPTION
**Pull request by lifeix**
Introduce Split Device Model feature for ACRN. Now we only implement pass through
PCI(e) device of it.

Tracked-On: #4371
Signed-off-by: Li Fei1 <fei1.li@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by ClaudZhang1995** [comment]
hi @lifeix 
make doc fail . It is fail log:
New errors/warnings found, please fix them:
==============================================

/root/workspace/acrn-hypervisor-premerge-build-image/acrn-hypervisor/build/doc/rst/developer-guides/hld/hv-dev-passthrough.rst:158: WARNING: undefined label: _splite-device-model (if the link has no caption the label must precede a section header)

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: doc/developer-guides/hld/split-dm.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.. _splite-device-model:
```
Comment (gvancuts) [inline doc/developer-guides/hld/split-dm.rst]: Can we fix this doc reference?
```suggestion
.. _split-device-model:
```

File: doc/developer-guides/hld/hv-dev-passthrough.rst [hunk 2]
```diff
@@ -155,24 +155,10 @@ directly. EPT mapping is not built for these pages having MSI-X table.
...
-  BAR, MSI capability, and INTLINE/INTPIN.
-
-- For other access, device model
-  reads/writes physical configuration space on behalf of UOS. To do
-  this, device model is linked with lib pci access to access physical PCI
-  device.
+PCI configuration space could been accessed by PCI-compatible Configuration Mechanism
+(IO port 0xCF8/CFC) and PCI Express Enhanced Configuration Access Mechanism (PCI MMCONFIG).
+ACRN hypervisor will trap these PCI configuration space access and emulate it. You may refer
+to :ref:`splite-device-model` for detail.
```
Comment (gvancuts) [inline doc/developer-guides/hld/hv-dev-passthrough.rst]:
```suggestion
to :ref:`split-device-model` for detail.
```

File: doc/developer-guides/hld/split-dm.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+   :name: split-dm-architecture-overview3
+
+.. figure:: images/split-dm-image4.png
+   :align: center
+   :width: 900px
+   :name: split-dm-architecture-overview4
+
+   PCI Config space access in Post-launched VM
+
+In order to achieve this, we add a new pair of hypercall hcall_assign_pcidev and
```
Comment (gvancuts) [inline doc/developer-guides/hld/split-dm.rst]: Can you use the `doxygenfunction` tag to reference these APIs?

File: doc/developer-guides/hld/split-dm.rst (hunk 1 above)
Comment (lifeix) [reply doc/developer-guides/hld/split-dm.rst]: Thanks for comments. Will fix it.

File: doc/developer-guides/hld/split-dm.rst (hunk 3 above)
Comment (lifeix) [reply doc/developer-guides/hld/split-dm.rst]: Sure. Thanks.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by NanlinXie** [comment]
@JasonChenCJ , could you help to review this document? Thanks!

**Pull request merged**